packages/client/hmi-client/src/services/chart-settings.ts

```diff
@@ -44,3 +54,213 @@ export function updateChartSettingsBySelectedVariables(
 	return settings.filter((setting) => setting.id !== id);
 }
+
+export async function saveAnnotation(
+	annotationLayerSpec: LLMGeneratedChartAnnotation,
+	nodeId: string,
+	chartId: string
+) {
+	const annotation: ChartAnnotation = {
+		id: uuidv4(),
+		description: annotationLayerSpec.request,
+		nodeId,
+		outputId: '',
+		chartId,
+		layerSpec: annotationLayerSpec.layerSpec,
+		llmGenerated: true,
+		metadata: {}
+	};
+	const { data } = await API.post('/chart-annotations', annotation);
+	return data as ChartAnnotation;
+}
+
+export async function deleteAnnotation(annotationId: string) {
+	const { data } = await API.delete(`/chart-annotations/${annotationId}`);
+	return data;
+}
+
+export async function fetchAnnotations(nodeId: string) {
+	const { data } = await API.post(`/chart-annotations/search`, { nodeId });
+	return data as ChartAnnotation[];
+}
+
+export async function generateForecastChartAnnotation(
+	request: string,
+	timeField: string,
+	variables: string[],
+	options: Partial<ForecastChartOptions>
+): Promise<LLMGeneratedChartAnnotation> {
+	const axisTitle = {
+		x: options.xAxisTitle ?? '',
+		y: options.yAxisTitle ?? ''
+	};
+	const translateMap = _.pick(options.translationMap ?? {}, variables);
+	const prompt = `
+	  You are an agent who is an expert in Vega-Lite chart specs. Provide a Vega-Lite layer JSON object for the annotation that can be added to an existing chart spec to satisfy the provided user request.
+
+    - The Vega-Lite schema version you must use is https://vega.github.io/schema/vega-lite/v5.json.
+    - Assume that you don’t know the exact data points from the data.
+    - You must give me the single layer object that renders all the necessary drawing objects, including multiple layers within the top layer object if needed.
+    - When adding a label, also draw a constant line perpendicular to the axis to which you are adding the label.
+
+    Assuming you are adding the annotations to the following chart spec,
+    ---- Example Chart Spec Start -----
+    {
+      "data": {"url": "data/samples.csv"},
+      "transform": [
+        {
+          "fold": ["price", "cost", "tax", "profit"],
+          "as": ["variableField", "valueField"]
+        }
+      ],
+      "layer": [
+        {
+          "mark": "line",
+          "encoding": {
+            "x": {"field": "date", "type": "quantitative", "axis": {"title": "Day"}},
+            "y": {"field": "valueField", "type": "quantitative", "axis": {"title": "Dollars"}}
+          }
+        }
+      ]
+    }
+    ---- Example Chart Spec End -----
+    Here are some example requests and the answers:
+
+    Request:
+    At day 200, add a label 'important'
+    Answer:
+    {
+      "description": "At day 200, add a label 'important'",
+      "layer": [
+        {
+          "mark": {
+            "type": "rule",
+            "strokeDash": [4, 4]
+          },
+					"encoding": {
+						"x": { "datum": 200, "axis": { "title": ""} }
+					}
+        },
```

Comment:
FYI: there seem to be some quirky tab formatting going on here.

Reply:
Fixed it